cmd/git_utils.go

```diff
@@ -129,19 +137,55 @@ func GetGitInfo(config *RootCommandConfig) (GitInfo, error) {
 		}
 
 	}
+
 	if strings.Contains(value, noCommits) {
 		gitInfo.Branch = StringAfter(value, noCommits)
 	}
+
 	changesMade := false
-	outputLength := len(outputLines)
+	outputLength := len(statusOutputLines)
 
 	if outputLength > 1 {
 		changesMade = true
-
 	}
 	gitInfo.ChangesMade = changesMade
 
-	errMsg := ""
+	if gitInfo.Upstream == "" {
+		config.Debug.log("Failed to retrieve upstream from git status -sb. Looking via latest commit")
+		outputLines, err := RunGitBranchContains(config.LoggingConfig, gitInfo.Commit.SHA, config.ProjectDir, lineSeparator, config.Dryrun)
+		if err != nil {
```

Comment:
What happens when the command doesn't return an error, and doesn't return any remotes? This would mean that `ouputLines` is empty, and `ouputLines[0]` could fail? Maybe we can do this check for empty within the `RunGitBranchContains` function, and return an error.
Regarding `errMsg` , if this fails we add a message. Then we do a `git remote` and if that is succesful we get the url, but the `errMsg` is not empty so [Line 199](https://github.com/appsody/appsody/pull/959/files#diff-5675d25a846ed226e9ba4514bd7eef60R199) will still return an error for the whole `GetGitInfo` function.

Reply:
I've added a bool value, `noRemoteFound` that is set to true if no remote has been found and false if one has. Then at the end of the function, a message is appended to `errMsg` if `noRemoteFound` is still true.